validation_testing/deploy_infra.sh

```diff
@@ -1,10 +1,24 @@
-
 CONNECTOR_NAME=$1
 VALIDATION_TESTING_ROOT=$REPOSITORY_ROOT/validation_testing
 
-# upload connector jar to s3 and update yaml to s3 uri, redirect to /dev/null to not log the s3 path
-aws s3 cp $REPOSITORY_ROOT/athena-$CONNECTOR_NAME/target/athena-$CONNECTOR_NAME-2022.47.1.jar "$S3_JARS_BUCKET/" > /dev/null
-sed -i "s#CodeUri: \"./target/athena-$CONNECTOR_NAME-2022.47.1.jar\"#CodeUri: \"$S3_JARS_BUCKET/athena-$CONNECTOR_NAME-2022.47.1.jar\"#" $REPOSITORY_ROOT/athena-$CONNECTOR_NAME/athena-$CONNECTOR_NAME.yaml
+# ecr repository must be created and an image pushed before the cdk stack is deployed for the lambda to create successfully
+# get the AWS account ID from the current roll (for use in ECR repo name)
+ACCOUNT_ID="$(aws sts get-caller-identity --query Account --output text)"
+
+# create the ECR repository
+aws ecr create-repository --repository-name athena-federation-repository-$CONNECTOR_NAME --region us-east-1
+
+# push the ECR image for the connector to the ECR repository
+aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin $ACCOUNT_ID.dkr.ecr.us-east-1.amazonaws.com
+docker build -t athena-federation-repository-$CONNECTOR_NAME $REPOSITORY_ROOT/athena-$CONNECTOR_NAME
+docker tag athena-federation-repository-$CONNECTOR_NAME\:latest $ACCOUNT_ID.dkr.ecr.us-east-1.amazonaws.com/athena-federation-repository-$CONNECTOR_NAME\:latest
+docker push $ACCOUNT_ID.dkr.ecr.us-east-1.amazonaws.com/athena-federation-repository-$CONNECTOR_NAME\:latest
+
+# update the template to use the correct ImageUri
+sed -i "s|292517598671|$ACCOUNT_ID|g" "$REPOSITORY_ROOT/athena-$CONNECTOR_NAME/athena-$CONNECTOR_NAME.yaml"
```

Comment:
we're exposing an account number here; we can make this into a github secret

Reply:
That account number is already in all of our yaml templates. It's something I brought up with AppSec when initially discussing ECR deployment and they said its fine.

Reply:
This goes and replaces the default account ID (the one already exposed in yaml templates) with the secret one used by github actions (gotten from the credentials so never exposed)